pkg/icingadb/v1/customvar.go

```diff
@@ -117,11 +118,16 @@ func flattenCustomvars(ctx context.Context, g *errgroup.Group, cvs <-chan contra
 					flattened := flatten.Flatten(value, customvar.Name)
 
 					for flatname, flatvalue := range flattened {
-						var fv string
+						var fv types.String
 						if flatvalue == nil {
-							fv = "null"
+							fv = types.String{NullString: sql.NullString{String: "null", Valid: true}}
 						} else {
-							fv = fmt.Sprintf("%v", flatvalue)
+							_, isSlice := flatvalue.([]interface{})
+							if _, ok := flatvalue.(map[string]interface{}); ok || isSlice {
+								fv = types.String{NullString: sql.NullString{String: "", Valid: false}}
+							} else {
+								fv = types.String{NullString: sql.NullString{String: fmt.Sprintf("%v", flatvalue), Valid: true}}
+							}
```

Comment:
> > It looks weird mapping `nil` to `'null'`
> 
> That's not something introduced with this PR.

Yes and no. "weird" was referring to the combination of both `'null'` and `NULL` handling (just saw that my sentence didn't survive some rephrasing I did while writing it so I edited it so that it should hopefully be grammatically correct now). This PR adds the need to insert `NULL`, so it's questionable if it still makes sense mapping `nil` to `'null'` as this now necessitates some other magic value for actual `NULL`.

> > and any (including non-empty) []interface{} to map[string]interface{} as NULL here.
> 
> That's not quite correct. There can be neither `non-empty` `[]interface{}` nor `maps` as flatvalue. But if it did, that would be a bug in the `Flatten()` function.

Non-empty isn't really the point here. It's that it uses surprising values for representing `NULL` which wouldn't be the case if `Flatten()` just returned `map[string]types.String`. Is there a downside in doing that?

Reply:
>  Is there a downside in doing that?

No. But we still have to map `nil` values to `null` string, otherwise, you won't be able to filter by something `host.some_custom_vars=null`.

Reply:
`Flatten()` would then either add `types.String{}` or `types.String{Value: "null", Valid: true}` to the map it returns. Then it would be clear inside that function what's going on there instead of it returning some non-flat (= non-scalar, I guess) values that are magically mapped to `NULL` somewhere else.